sunbeam-python/tests/unit/sunbeam/plugins/test_validation.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from pathlib import Path
+from unittest.mock import AsyncMock, Mock, patch
+
+import click
+import pytest
+
+from sunbeam.plugins.validation import plugin as validation_plugin
+
+
+@pytest.fixture(autouse=True)
+def mock_run_sync(mocker):
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        loop = asyncio.new_event_loop()
+
+    def run_sync(coro):
+        return loop.run_until_complete(coro)
+
+    mocker.patch("sunbeam.plugins.validation.plugin.run_sync", run_sync)
+    yield
+    loop.close()
+
+
+@pytest.fixture()
+def cclient():
+    with patch("sunbeam.plugins.interface.v1.base.Client") as p:
+        yield p
+
+
+@pytest.fixture()
+def jhelper():
+    yield AsyncMock()
+
+
+@pytest.fixture()
+def tfhelper():
+    yield Mock(path=Path())
+
+
+@pytest.fixture()
+def validationplugin():
+    with patch("sunbeam.plugins.validation.plugin.ValidationPlugin") as p:
+        yield p
+
+
+class TestValidatorFunction:
```

Comment:
What kind of test are you thinking of? the validation plugin is a subclass of `OpenStackControlPlanePlugin`, and most have already be tested by `test_openstack.py`, and we didn't introduce new code for interacting with plugin, except for the actions, and click commands and group (testing it may be hard, though there are some library to test cli)

Reply:
we can extract all the functions from the `click.command` and test those function though, sounds good ?